x/evm/watcher/watcher.go

```diff
@@ -336,9 +374,52 @@ func (w *Watcher) Commit() {
 	if tmtypes.UploadDelta {
 		return
 	}
-	w.dispatchJob(func() {
-		w.commitBatch(batch)
-	})
+
+	if GetEnableAsyncCommit() {
+		shouldPersist := w.IsShouldPersist()
+		height := w.height
+		w.dispatchJob(func() {
+			acBatch, noACBatch := ClassifyWatchMessageType(batch)
+			w.acProcessor.BatchSet(acBatch)
+			if shouldPersist {
+				// move curMsgCache to commitlist
+				w.acProcessor.MoveToCommitList(int64(height))
+				w.commitBatch(noACBatch)
+				w.AsyncCommitBatch()
+			} else {
+				w.commitBatch(noACBatch)
```

Comment:
Seems we can move `w.commitBath` out of the code block ,then remove the `else` code block.

Reply:
ok